test/subtype.jl

```diff
@@ -796,6 +800,11 @@ function test_intersection()
                    (@UnionAll X<:(@UnionAll T Tuple{T,T}) Ref{X}))
     @testintersect(Ref{@UnionAll T @UnionAll S Tuple{T,S}},
                    Ref{@UnionAll T Tuple{T,T}}, Bottom)
+
+    @testintersect(Tuple{T,T} where T<:Union{UpperTriangular, UnitUpperTriangular},
+                   Tuple{AbstractArray{T,N}, AbstractArray{T,N}} where N where T,
```

Comment:
Yeah, that's a good question. I think that having the first variable be the outermost would make the most sense for `where T, S`, though if both syntaxes are supported (i.e. `,` and nested), having the orders match would probably be best. Writing `where S<:Vector{T}, T<:Real` seems okay to me.